samcli/lib/list/resources/resource_mapping_producer.py

```diff
@@ -0,0 +1,80 @@
+"""
+in progress
+"""
+from typing import Optional, Any
+import dataclasses
+import logging
+import yaml
+import click
+
+from botocore.exceptions import ClientError, NoCredentialsError
+from samtranslator.translator.managed_policy_translator import ManagedPolicyLoader
+from samtranslator.translator.arn_generator import NoRegionFound
+
+from samcli.commands.list.exceptions import SamListLocalResourcesNotFoundError, SamListUnknownClientError
+from samcli.lib.list.list_interfaces import Producer
+from samcli.lib.list.resources.resources_def import ResourcesDef
+from samcli.lib.translate.sam_template_validator import SamTemplateValidator
+from samcli.lib.providers.sam_stack_provider import SamLocalStackProvider
+from samcli.lib.translate.translate_utils import _read_sam_file
+from samcli.lib.translate.exceptions import InvalidSamDocumentException
+from samcli.commands.local.cli_common.user_exceptions import InvalidSamTemplateException
+from samcli.commands.exceptions import UserException
+
+LOG = logging.getLogger(__name__)
+
+
+class ResourceMappingProducer(Producer):
+    def __init__(
+        self, stack_name, output, region, profile, template_file, cloudformation_client, iam_client, mapper, consumer
+    ):
+        self.stack_name = stack_name
+        self.output = output
+        self.region = region
+        self.profile = profile
+        self.template_file = template_file
+        self.cloudformation_client = cloudformation_client
+        self.iam_client = iam_client
+        self.mapper = mapper
+        self.consumer = consumer
+
+    def get_translated_dict(self, template_file_dict) -> Optional[Any]:
+        try:
+            validator = SamTemplateValidator(
+                template_file_dict, ManagedPolicyLoader(self.iam_client), profile=self.profile, region=self.region
+            )
+            translated_dict = yaml.load(validator.get_translated_template(), Loader=yaml.FullLoader)
```

Comment:
I wonder if we should really be making use of the validator here 🤔 . It seems what we really want is to just get the translated template. So maybe we should extract that logic into somewhere else where both this code and the validator code make use of it